### PR TITLE
design basic skeleton for tweet placeholder

### PR DESCRIPTION
**Pull request by surajsharma**
resolves: https://github.com/holdex/holdex-venture-studio/issues/324

what we have here is an attempt, tailwind is again acting funky here so your feedback is most welcome!

Time  taken: 15 mins

### CODE REVIEW

**Comment by predrag-us** [comment]
@surajsharma 

Can you please attach a URL of where the solution can be located? Thank you!

**Comment by surajsharma** [comment]
@predrag-us thanks for your help, 

- if you fetch the `issue-324/tweet-skeleton` branch you can see my attempt on `http://localhost:3000/c/elysio/testing-multiple-tweets` 

- head over to https://holdex-venture-studio-git-issue-324-t-b40f31-holdex-accelerator.vercel.app/c/elysio/testing-multiple-tweets to see it in action

**Comment by predrag-us** [comment]
@surajsharma

The skeleton should resemble the looks of the loaded component, as can [be seen here](https://www.figma.com/file/39MGGVIGQ1bL3FgVzuXEdC/UI-Design?type=design&node-id=2628%3A39215&t=noK8WqIBb6M8yXiM-1).  

In the example above, I've converted the data that awaits fetching into a skeleton but kept the static elements intact. 

This doesn't necessarily mean that you need to make them identical (especially since it's an embed so it may be harder), but try to add some distinctive features (e.g. avatar) to make them "feel" the same, so the visitor knows what to expect. 

Also, do you understand the note since you're using different colors?

**Comment by surajsharma** [comment]
@predrag-us thanks for the comment, i will try and add some distinctive feel here. Adjusting colors as well.

**Comment by surajsharma** [comment]
https://github.com/holdex/holdex-venture-studio/assets/5298624/ecab069d-053a-4ce3-aa4e-1dd5a5682519



@predrag-us i was able to achieve a basic skeleton per your guidance, see the video attached, please review

**Comment by georgeciubotaru** [comment]
@surajsharma To get your PR reviewed please:
1. Mark as `Ready for review`
2. Assign the person who should review
3. Add reviewers

**Comment by surajsharma** [comment]
@georgeciubotaru totally forgot to do that, done now thanks!

**Comment by predrag-us** [comment]
@surajsharma 

Good progress, Suraj! 🙌

Some of the problems I mentioned previously still persist. 

You should keep the elements sizes, gaps, margins, and paddings from the fully-rendered component and apply those to the skeleton. 

Based on what you've already done, I've prepared an example of how I think the skeleton should look. As you can tell, you're nearly there, the difference is in the details. 

I've kept everything as in the fully-rendered component, removed some of the elements based on your work, and swapped content that awaits fetching with skeleton elements. 

Even when designing skeleton elements, our aim is to be pixel-perfect and provide the best possible experience to our visitors.

You can implement what I prepared and call it a day, but I would recommend you to spend some time and understand how and why behind the solution, so we don't prolong this kind of issue in the future. 

Please try to finish working on this issue in a timely manner so you can start working on more important issues. 

[Design URL](https://www.figma.com/file/39MGGVIGQ1bL3FgVzuXEdC/UI-Design?type=design&node-id=2739%3A39127&t=Xyb7X6amdP1yL4e7-1) 

CleanShot 2023-06-12 at 10 06 31@2x

**Comment by surajsharma** [comment]
@predrag-us thank you for your review, i will learn from, and iterate upon your inputs and see if i can make it more like the tweet, **streched on other tasks so request some time here**. 

pls also note that i can just use your iteration and call it a day but i'm thinking about introducing static elements, i had a couple of questions regarding this:

1_ is it ok to put text in a skeleton? such as "loading..." or "copy link?"
2_ is it ok to put static icons (such as the heart or quote icon) into the skeleton? i have not seen this done elsewhere


i'll see if i can come up with more adjustments.

cc: @georgeciubotaru

**Comment by predrag-us** [comment]
@surajsharma It's a skeleton so I recommend keeping it simple yet recognizable. We don't need to spend time on additional elements IMO.

**Comment by georgeciubotaru** [comment]
@surajsharma Let me know when I can do the code review

**Comment by surajsharma** [comment]
@georgeciubotaru pls feel free to review now, I've followed @predrag-us's new design and i believe more effort here could prove wasteful

File: src/lib/components/BodyRenderer/Blocks/Tweet/components/twitter-layout/Tweet/TweetSkeleton.svelte [hunk 1]
```diff
@@ -6,19 +6,84 @@
 	export let simple = false;
 
 	let theme = getContext('theme');
+
+	const twitterSrc = 'https://storage.googleapis.com/stage-holdex-public/assets/twitter.png';
```
[the remark this answers is not in the document]
Comment (surajsharma) [reply src/lib/components/BodyRenderer/Blocks/Tweet/components/twitter-layout/Tweet/TweetSkeleton.svelte:10]: note: there is a bit of a catch here, in that importing directly from static folder is not available/doesn't work,
instead the workaround features using vite imperatively like so:

`const twitter = new URL('/twitter.png', import.meta.url).href;`


which makes the skeleton display a broken image until it is loaded. 
this likely won't be a problem once the page is SSRd, but i know of no better way to achieve this as of yet.

File: src/lib/components/BodyRenderer/Blocks/Tweet/components/twitter-layout/Tweet/TweetSkeleton.svelte [hunk 2]
```diff
@@ -7,32 +7,27 @@
 
 	let theme = getContext('theme');
 
-	const twitterSrc = 'https://storage.googleapis.com/stage-holdex-public/assets/twitter.png';
+	const twitter = new URL('/twitter.png', import.meta.url).href;
```
Comment (georgeciubotaru) [inline src/lib/components/BodyRenderer/Blocks/Tweet/components/twitter-layout/Tweet/TweetSkeleton.svelte:10]: @surajsharma What error do you have by using `/twitter.png` directly in img src?

Comment (surajsharma) [reply src/lib/components/BodyRenderer/Blocks/Tweet/components/twitter-layout/Tweet/TweetSkeleton.svelte:10]: works!

File: src/lib/components/BodyRenderer/Blocks/Tweet/components/twitter-layout/Tweet/TweetSkeleton.svelte [hunk 3]
```diff
@@ -3,22 +3,78 @@
...
+				>
+					<span class="author animation animate overflow-hidden ${theme}" title="name" />
+					<span class="username animation animate overflow-hidden ${theme}" title="username" />
+				</span>
+			</div>
+			<img
+				class="icon w-[1.25em] h-[1.25em] object-contain twicon brand exclude"
+				title="View on Twitter"
+				alt="twitter"
+				src={'/twitter.png'}
```
Comment (georgeciubotaru) [inline src/lib/components/BodyRenderer/Blocks/Tweet/components/twitter-layout/Tweet/TweetSkeleton.svelte:25]:
```suggestion
				src="/twitter.png"
```
@surajsharma Why not that way?

**Comment by georgeciubotaru** [comment]
@surajsharma As an additional task, you can refactor the other image URLs and move them in the repo (better to group them in a folder)

**Comment by surajsharma** [comment]
@georgeciubotaru have refactored the image urls but be advised that these aren't being used, instead i am using icons exported as svg from `/Users/admin/evenzero/jobs/holdex/holdex-venture-studio/src/lib/components/BodyRenderer/Blocks/Tweet/components/twitter-layout/Icons/index.ts` as you had suggested here https://github.com/holdex/holdex-venture-studio/pull/317#issuecomment-1562941041

**Review by georgeciubotaru — APPROVED** [review]

**Pull request merged**